packages/public-api/src/middleware/common.ts

```diff
@@ -1,11 +1,34 @@
-import { ValidationError } from 'joi';
+import { ZodError } from 'zod';
 import createCorsMiddleware from 'cors';
-import express, { Request, Response, NextFunction } from 'express';
+import { Request, Response, NextFunction } from 'express';
 import { NotFoundError, respondWithError } from './response';
 
-// TODO: add whitelist origins
-export const cors = createCorsMiddleware({ maxAge: 1800 });
-export const parseJson = express.json();
+const whitelist = {
+  production: ['https://gradejs.com'],
+  staging: ['https://staging.gradejs.com'],
+  development: ['http://localhost:3000'],
+};
+
+const getNodeEnv = () => {
```

Comment:
I believe we already have such function in the `shared` package

Reply:
The one in `shared` appears to be insufficient for this use. Do you mind replacing that function with this implementation?

Reply:
Yeah, that should work